corpus/segments.py

```diff
@@ -261,6 +262,43 @@ def run(self):
                 f.writelines(lines[start:end])
 
 
+class DynamicSplitSegmentFileJob(Job):
+
+    """
+    This split the segments to concurrent many shares.
+    It is a variant to the existing SplitSegmentFileJob, which requires a delayed variable for argument concurrent.
+    """
+
+    def __init__(self, segment_file: tk.Union[tk.Path, str], concurrent: tk.Delayed):
+        """
+        :param tk.Union[tk.Path, str] segment_file: segment file
+        :param tk.Delayed concurrent: number of splits
+        """
+        self.segment_file = segment_file
+        self.concurrent = concurrent
+        self.out_split_dir = self.output_path("split", directory=True)
+
+    def tasks(self):
+        yield Task("run", resume="run", mini_task=True)
+
+    def run(self):
+        with open(tk.uncached_path(self.segment_file), "rt") as f:
```

Comment:
use `util.uopen` instead of `open`. see other usages

Reply:
Did we already put this in the guidelines file? If not I will add this

Reply:
ok thanks

Reply:
should I change correspondingly for all other Jobs in that file which has this issue?

Reply:
No this is not needed, I already planned to do it for all Jobs at once, it is just that new submissions should already follow this.